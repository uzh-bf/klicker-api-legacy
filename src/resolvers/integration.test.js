const request = require('supertest')
const mongoose = require('mongoose')

const server = require('../app')
const { initializeDb } = require('../lib/test/setup')

process.env.NODE_ENV = 'test'

describe('integration', () => {
  let authCookie

  beforeAll(async () => {
    await initializeDb({ mongoose, email: 'testIntegration@bf.uzh.ch', shortname: 'integr' })
  })

  afterAll(async () => {
    await mongoose.disconnect()
    authCookie = null
  })

  describe('login', () => {
    it('works', async () => {
      // send a login request
      const response = await request(server)
        .post('/graphql')
        .send({
          query: `
            mutation LoginUser($email: String!, $password: String!) {
              login(email: $email, password: $password) {
                email
                shortname
              }
            }
          `,
          operationName: 'LoginUser',
          variables: {
            email: 'testIntegration@bf.uzh.ch',
            password: 'somePassword',
          },
        })

      // save the authorization cookie
      authCookie = response.header['set-cookie']

      expect(response.body.data).toMatchSnapshot()
      expect(authCookie.length).toEqual(1)
    })
  })
})
