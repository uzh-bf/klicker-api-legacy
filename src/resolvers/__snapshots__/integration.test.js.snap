// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration login works 1`] = `
Object {
  "login": Object {
    "email": "testIntegration@bf.uzh.ch",
    "shortname": "integr",
  },
}
`;
